### ISSUE TITLE
Differing behavior between ptk and readline backend

### ISSUE THREAD

**Issue by cafehaine** [xonsh/xonsh#3801]
## Expected Behavior
The ptk and readline backend shouldn't show any obvious difference when running.

## Current Behavior
The modifiers are reset when changing colors, but only for the ptk backend.

## Steps to Reproduce

```python
import xonsh.tools
xonsh.tools.print_color("{BOLD_GREEN}green {RED}red")
```
This example will print both `green` and `red` in bold (which is the intended behavior) with the readline backend, but only `green` will be bold when using the ptk backend.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by bobhy** [comment]
Agreed different behavior should be reconciled, but which is the "correct" behavior? Is there an external tiebreaker you can point us at?

**Comment by scopatz** [comment]
I think in most of the examples that @cafehaine has show, it seems like PTK is throwing out the non-color formatting info

**Comment by cafehaine** [comment]
IMO, the PTK backend is wrong discarding the modifiers.

I've tried to look into it, but it seems that we would need to keep all the modifiers from one "token" to the next, so it might be quite complex to fix this

**Comment by scopatz** [comment]
Maybe this is something to bring up with PTK?

**Comment by cafehaine** [comment]
I'm not sure about that, it might be on us to make sure it works as we intend it to, as PTK isn't really made for this specific use case

Their "token" system is built to reset everything once exiting a token, which is a sensible behavior, but not the one we want to achieve
